torchao/dtypes/nf4tensor.py

```diff
@@ -186,8 +186,7 @@ def __new__(
             tensor_meta.original_shape,
             tensor_meta.original_strides,
             tensor_meta.storage_offset,
-            # Picked some floating dtype, but we need dtype extensibility
-            dtype=torch.float8_e5m2fnuz,
+            dtype=tensor_meta.dtype, # Dtype broadcast to the world matches input
```

Comment:
Input during construction. Does this mean `.to(torch.float32)` should change this `dtype` as well and also cause different results when someone calls `dequantize` or somesuch?

Reply:
It does make sense to have this to figure out the dtype of the lookup table and better cache the dequantize operation (which could also be mitigated with multiple lookup tables or kernels).